scrunch/datasets.py

```diff
@@ -1456,15 +1454,11 @@ def __repr__(self):
     def __str__(self):
         return self.name
 
-    _categories = None
-
     @property
     def categories(self):
         if self.resource.body['type'] not in self.CATEGORICAL_TYPES:
             raise TypeError("Variable of type %s do not have categories" % self.resource.body.type)
-        if self._categories is None:
-            self._categories = CategoryList(self.resource)
-        return self._categories
+        return CategoryList._from(self.resource)
```

Comment:
the reason for that is related to the different approach of accessing the attributes with the new class. Formerly you were using the `fill` method which was directly setting the `Category` attributes (which should be forbidden by now). Changing their attributes with the `edit` method had no effect on the instances in the `CategoryList` so basically there was a conflict between having up to date `Category` instances in the `CategoryList` (`test_edit_category`) with a single intantiation of it (`test_instance_is_reused`).